Failed build on M1 pro Sonoma with rust 1.70.0

```
make
cargo build --release
   Compiling autocfg v1.1.0
   Compiling serde v1.0.160
   Compiling proc-macro2 v1.0.56
   Compiling unicode-ident v1.0.8
   Compiling cc v1.0.79
   Compiling quote v1.0.26
   Compiling syn v1.0.109
   Compiling memchr v2.5.0
   Compiling pkg-config v0.3.26
   Compiling hashbrown v0.12.3
   Compiling erased-serde v0.3.25
   Compiling indexmap v1.9.3
   Compiling lua-src v544.0.1
   Compiling luajit-src v210.4.5+resty2cf5186
   Compiling num-traits v0.2.15
   Compiling winnow v0.4.1
   Compiling mlua v0.8.8
   Compiling bstr v0.2.17
   Compiling once_cell v1.17.1
   Compiling rustc-hash v1.1.0
   Compiling toml_datetime v0.6.1
   Compiling serde_spanned v0.6.1
   Compiling toml_edit v0.19.8
   Compiling mlua_derive v0.8.0
   Compiling toml v0.7.3
   Compiling rsync-nvim v0.1.0 (/Users/xxx/tmp/rsync.nvim)
error: linking with `cc` failed: exit status: 1
  |
  = note: LC_ALL="C" PATH="/Users/xxx/Code/dotfiles/.asdf/installs/rust/1.70.0/toolchains/1.70.0-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/bin:/Users/xxx/.asdf/installs/rust/1.70.0/bin:/opt/homebrew/opt/tcl-tk/bin:/opt/homebrew/opt/openssl@3/bin:/opt/homebrew/opt/sqlite/bin:/opt/homebrew/opt/bzip2/bin:/Users/xxx/.asdf/bin:/Users/xxx/.orbstack/bin:/Users/xxx/.asdf/shims:/opt/homebrew/libexec/gnubin:/opt/homebrew/share/luarocks/bin:/Users/xxx/.poetry/bin:/Users/xxx/.cargo/bin:/Users/xxx/.local/bin:/bin:/sbin:/usr/bin:/usr/sbin:/Users/xxx/.fzf/bin:/Users/xxx/.oh-my-zsh/custom/plugins/fzf-zsh-plugin/bin:/opt/homebrew/opt/fzf/bin:/opt/local/bin:/opt/local/sbin:/opt/homebrew/opt/gcc/bin:/opt/homebrew/opt/qt/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/opt/homebrew/libexec/gnubin:/opt/homebrew/share/luarocks/bin:/usr/local/bin:/usr/local/sbin" VSLANG="1033" ZERO_AR_DATE="1" "cc" "-Wl,-exported_symbols_list,/var/folders/z0/rppzdldd3r99k2k5s7xfg8nh0000gn/T/rustcFrlWqz/list" "-arch" "arm64" "/var/folders/z0/rppzdldd3r99k2k5s7xfg8nh0000gn/T/rustcFrlWqz/symbols.o" "/Users/xxx/tmp/rsync.nvim/target/release/deps/rsync_nvim.rsync_nvim.b3f25e6e-cgu.0.rcgu.o" "/Users/xxx/tmp/rsync.nvim/target/release/deps/rsync_nvim.rsync_nvim.b3f25e6e-cgu.1.rcgu.o" "/Users/xxx/tmp/rsync.nvim/target/release/deps/rsync_nvim.rsync_nvim.b3f25e6e-cgu.10.rcgu.o" "/Users/xxx/tmp/rsync.nvim/target/release/deps/rsync_nvim.rsync_nvim.b3f25e6e-cgu.11.rcgu.o" "/Users/xxx/tmp/rsync.nvim/target/release/deps/rsync_nvim.rsync_nvim.b3f25e6e-cgu.12.rcgu.o" "/Users/xxx/tmp/rsync.nvim/target/release/deps/rsync_nvim.rsync_nvim.b3f25e6e-cgu.13.rcgu.o" "/Users/xxx/tmp/rsync.nvim/target/release/deps/rsync_nvim.rsync_nvim.b3f25e6e-cgu.14.rcgu.o" "/Users/xxx/tmp/rsync.nvim/target/release/deps/rsync_nvim.rsync_nvim.b3f25e6e-cgu.15.rcgu.o" "/Users/xxx/tmp/rsync.nvim/target/release/deps/rsync_nvim.rsync_nvim.b3f25e6e-cgu.2.rcgu.o" "/Users/xxx/tmp/rsync.nvim/target/release/deps/rsync_nvim.rsync_nvim.b3f25e6e-cgu.3.rcgu.o" "/Users/xxx/tmp/rsync.nvim/target/release/deps/rsync_nvim.rsync_nvim.b3f25e6e-cgu.4.rcgu.o" "/Users/xxx/tmp/rsync.nvim/target/release/deps/rsync_nvim.rsync_nvim.b3f25e6e-cgu.5.rcgu.o" "/Users/xxx/tmp/rsync.nvim/target/release/deps/rsync_nvim.rsync_nvim.b3f25e6e-cgu.6.rcgu.o" "/Users/xxx/tmp/rsync.nvim/target/release/deps/rsync_nvim.rsync_nvim.b3f25e6e-cgu.7.rcgu.o" "/Users/xxx/tmp/rsync.nvim/target/release/deps/rsync_nvim.rsync_nvim.b3f25e6e-cgu.8.rcgu.o" "/Users/xxx/tmp/rsync.nvim/target/release/deps/rsync_nvim.rsync_nvim.b3f25e6e-cgu.9.rcgu.o" "/Users/xxx/tmp/rsync.nvim/target/release/deps/rsync_nvim.1g422nfgh91ih13i.rcgu.o" "-L" "/Users/xxx/tmp/rsync.nvim/target/release/deps" "-L" "/Users/xxx/Code/dotfiles/.asdf/installs/rust/1.70.0/toolchains/1.70.0-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib" "/Users/xxx/tmp/rsync.nvim/target/release/deps/libtoml-4f8d4084a8500585.rlib" "/Users/xxx/tmp/rsync.nvim/target/release/deps/libtoml_edit-58b3d17e8cb2a579.rlib" "/Users/xxx/tmp/rsync.nvim/target/release/deps/libserde_spanned-61b938f7c9706d37.rlib" "/Users/xxx/tmp/rsync.nvim/target/release/deps/libindexmap-100fce0537b33a7c.rlib" "/Users/xxx/tmp/rsync.nvim/target/release/deps/libhashbrown-3ddb46b5aa02d1e0.rlib" "/Users/xxx/tmp/rsync.nvim/target/release/deps/libwinnow-a8e770cab089e272.rlib" "/Users/xxx/tmp/rsync.nvim/target/release/deps/libtoml_datetime-56be782acb6fe45b.rlib" "/Users/xxx/tmp/rsync.nvim/target/release/deps/libmlua-76ae36068b5adf9d.rlib" "/Users/xxx/tmp/rsync.nvim/target/release/deps/liberased_serde-8c8ed8a7e6f324fd.rlib" "/Users/xxx/tmp/rsync.nvim/target/release/deps/libonce_cell-f98a4e84f3ae5964.rlib" "/Users/xxx/tmp/rsync.nvim/target/release/deps/libserde-4cb75b35e8c72b04.rlib" "/Users/xxx/tmp/rsync.nvim/target/release/deps/librustc_hash-5ec9948dfc745fe7.rlib" "/Users/xxx/tmp/rsync.nvim/target/release/deps/libnum_traits-9a71963d467d10fe.rlib" "/Users/xxx/tmp/rsync.nvim/target/release/deps/libbstr-240152852bf84d18.rlib" "/Users/xxx/tmp/rsync.nvim/target/release/deps/libmemchr-71bec0c05e2e34d8.rlib" "/Users/xxx/Code/dotfiles/.asdf/installs/rust/1.70.0/toolchains/1.70.0-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libstd-2ca50e2550a11ed4.rlib" "/Users/xxx/Code/dotfiles/.asdf/installs/rust/1.70.0/toolchains/1.70.0-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libpanic_unwind-a0b15874ec5099e6.rlib" "/Users/xxx/Code/dotfiles/.asdf/installs/rust/1.70.0/toolchains/1.70.0-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libobject-dca0dfe66399d350.rlib" "/Users/xxx/Code/dotfiles/.asdf/installs/rust/1.70.0/toolchains/1.70.0-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libmemchr-139c697183de5a45.rlib" "/Users/xxx/Code/dotfiles/.asdf/installs/rust/1.70.0/toolchains/1.70.0-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libaddr2line-ab400bc75f06afe5.rlib" "/Users/xxx/Code/dotfiles/.asdf/installs/rust/1.70.0/toolchains/1.70.0-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libgimli-a21366080735b75e.rlib" "/Users/xxx/Code/dotfiles/.asdf/installs/rust/1.70.0/toolchains/1.70.0-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/librustc_demangle-59d24b42d35c0fa1.rlib" "/Users/xxx/Code/dotfiles/.asdf/installs/rust/1.70.0/toolchains/1.70.0-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libstd_detect-4fc124adb92a6b8c.rlib" "/Users/xxx/Code/dotfiles/.asdf/installs/rust/1.70.0/toolchains/1.70.0-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libhashbrown-0d73503e669513a0.rlib" "/Users/xxx/Code/dotfiles/.asdf/installs/rust/1.70.0/toolchains/1.70.0-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libminiz_oxide-1b85702a684cf585.rlib" "/Users/xxx/Code/dotfiles/.asdf/installs/rust/1.70.0/toolchains/1.70.0-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libadler-a24e569f9e1f1731.rlib" "/Users/xxx/Code/dotfiles/.asdf/installs/rust/1.70.0/toolchains/1.70.0-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/librustc_std_workspace_alloc-76480388111a6770.rlib" "/Users/xxx/Code/dotfiles/.asdf/installs/rust/1.70.0/toolchains/1.70.0-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libunwind-c16d55155160ec76.rlib" "/Users/xxx/Code/dotfiles/.asdf/installs/rust/1.70.0/toolchains/1.70.0-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libcfg_if-83fe9edd4bff95ef.rlib" "/Users/xxx/Code/dotfiles/.asdf/installs/rust/1.70.0/toolchains/1.70.0-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/liblibc-c469bd47c3550b1c.rlib" "/Users/xxx/Code/dotfiles/.asdf/installs/rust/1.70.0/toolchains/1.70.0-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/liballoc-ad1eeb16b5006415.rlib" "/Users/xxx/Code/dotfiles/.asdf/installs/rust/1.70.0/toolchains/1.70.0-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/librustc_std_workspace_core-6886eedc65dccb14.rlib" "/Users/xxx/Code/dotfiles/.asdf/installs/rust/1.70.0/toolchains/1.70.0-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libcore-238d26edba85c951.rlib" "/Users/xxx/Code/dotfiles/.asdf/installs/rust/1.70.0/toolchains/1.70.0-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libcompiler_builtins-8940904d1c2f93b4.rlib" "-lSystem" "-lc" "-lm" "-L" "/Users/xxx/Code/dotfiles/.asdf/installs/rust/1.70.0/toolchains/1.70.0-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib" "-o" "/Users/xxx/tmp/rsync.nvim/target/release/deps/librsync_nvim.dylib" "-Wl,-dead_strip" "-dynamiclib" "-Wl,-dylib" "-nodefaultlibs"
  = note: Undefined symbols for architecture arm64:
            "_luaL_callmeta", referenced from:
                mlua::ffi::lua51::compat::luaL_tolstring::h2b713157f0508532 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.5.rcgu.o)
            "_luaL_error", referenced from:
                mlua::ffi::lua51::compat::lua_rotate::hdf6ac2ba506c53a9 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.5.rcgu.o)
                mlua::ffi::lua51::compat::luaL_traceback::h8bc46a5d3f85f45a in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.5.rcgu.o)
                mlua::ffi::lua51::compat::luaL_tolstring::h2b713157f0508532 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.5.rcgu.o)
                mlua::lua::Lua::create_callback::call_callback::hf62fd768a32fb2f3 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                mlua::lua::Lua::init_from_ptr::_$u7b$$u7b$closure$u7d$$u7d$::do_call::h9b88f980dcf859c5 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                mlua::util::protect_lua_closure::do_call::h418a40b0275da235 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                mlua::util::callback_error::he749de5b9e226a6a in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                ...
            "_luaL_getmetafield", referenced from:
                mlua::ffi::lua51::compat::luaL_tolstring::h2b713157f0508532 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.5.rcgu.o)
            "_luaL_ref", referenced from:
                mlua::util::protect_lua_closure::do_call::hf84c9facd2040c28 (.llvm.3734580924993591236) in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
            "_lua_checkstack", referenced from:
                mlua::function::Function::call::h20041b423bb18ae9 in rsync_nvim.rsync_nvim.b3f25e6e-cgu.11.rcgu.o
                mlua::table::Table::set::hff356e86ce69b428 in rsync_nvim.rsync_nvim.b3f25e6e-cgu.3.rcgu.o
                mlua::table::Table::raw_set::hd238aea89fcbe55e in rsync_nvim.rsync_nvim.b3f25e6e-cgu.3.rcgu.o
                mlua::table::Table::raw_set::hd3243c56630bce63 in rsync_nvim.rsync_nvim.b3f25e6e-cgu.3.rcgu.o
                mlua::lua::Lua::entrypoint1::h1a2161b5a70a9116 in rsync_nvim.rsync_nvim.b3f25e6e-cgu.4.rcgu.o
                mlua::lua::Lua::create_string::ha18e72160fc63138 in rsync_nvim.rsync_nvim.b3f25e6e-cgu.4.rcgu.o
                mlua::lua::Lua::create_string::hf30d545905e2b48b in rsync_nvim.rsync_nvim.b3f25e6e-cgu.4.rcgu.o
                ...
            "_lua_concat", referenced from:
                mlua::ffi::lua51::compat::compat53_findfield::h548718bee81eb3ed in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.5.rcgu.o)
                mlua::ffi::lua51::compat::luaL_traceback::h8bc46a5d3f85f45a in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.5.rcgu.o)
            "_lua_createtable", referenced from:
                mlua::lua::Lua::create_table_with_capacity::h5894c7c379a161e3 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                mlua::util::protect_lua_closure::do_call::h6b25b093f024bdf9 (.llvm.3734580924993591236) in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                mlua::serde::init_metatables::do_call::hdd942496a810825a (.llvm.11307603184714879020) in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.13.rcgu.o)
            "_lua_error", referenced from:
                mlua::lua::Lua::entrypoint1::h1a2161b5a70a9116 in rsync_nvim.rsync_nvim.b3f25e6e-cgu.4.rcgu.o
                mlua::lua::Lua::create_callback::call_callback::hf62fd768a32fb2f3 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                mlua::util::callback_error::he749de5b9e226a6a in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                mlua::util::init_error_registry::error_tostring::h8e2e0838af2d7cf3 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                mlua::util::init_error_registry::destructed_error::hea4da3b5b9b49f41 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
            "_lua_getinfo", referenced from:
                mlua::ffi::lua51::compat::luaL_traceback::h8bc46a5d3f85f45a in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.5.rcgu.o)
            "_lua_getmetatable", referenced from:
                mlua::table::Table::set::hff356e86ce69b428 in rsync_nvim.rsync_nvim.b3f25e6e-cgu.3.rcgu.o
                mlua::util::pop_error::h04ae61cb9f92798f in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                mlua::util::get_gc_userdata::hb89e34bf79757476 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                mlua::util::error_traceback::he68037b9b5234d0f in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                std::panicking::try::hc905dbe356661666 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.12.rcgu.o)
            "_lua_getstack", referenced from:
                mlua::ffi::lua51::compat::luaL_traceback::h8bc46a5d3f85f45a in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.5.rcgu.o)
            "_lua_gettop", referenced from:
                mlua::function::Function::call::h20041b423bb18ae9 in rsync_nvim.rsync_nvim.b3f25e6e-cgu.11.rcgu.o
                mlua::util::protect_lua_closure::h4ebbacc47b197a8d in rsync_nvim.rsync_nvim.b3f25e6e-cgu.13.rcgu.o
                mlua::util::protect_lua_closure::do_call::h734df75e92650b3f (.llvm.453382158249156745) in rsync_nvim.rsync_nvim.b3f25e6e-cgu.13.rcgu.o
                mlua::table::Table::set::hff356e86ce69b428 in rsync_nvim.rsync_nvim.b3f25e6e-cgu.3.rcgu.o
                mlua::table::Table::raw_set::hd238aea89fcbe55e in rsync_nvim.rsync_nvim.b3f25e6e-cgu.3.rcgu.o
                mlua::table::Table::raw_set::hd3243c56630bce63 in rsync_nvim.rsync_nvim.b3f25e6e-cgu.3.rcgu.o
                mlua::lua::Lua::entrypoint1::h1a2161b5a70a9116 in rsync_nvim.rsync_nvim.b3f25e6e-cgu.4.rcgu.o
                ...
            "_lua_insert", referenced from:
                mlua::ffi::lua51::compat::compat53_findfield::h548718bee81eb3ed in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.5.rcgu.o)
                mlua::ffi::lua51::compat::lua_rotate::hdf6ac2ba506c53a9 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.5.rcgu.o)
                mlua::lua::Lua::create_callback::call_callback::hf62fd768a32fb2f3 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                mlua::lua::Lua::init_from_ptr::_$u7b$$u7b$closure$u7d$$u7d$::do_call::h9b88f980dcf859c5 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                mlua::util::protect_lua_call::h5f86f7c4ef732698 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                mlua::util::protect_lua_closure::h67599b8c8d98bd27 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                mlua::util::protect_lua_closure::had1f2c3e9ee6b79d in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                ...
            "_lua_isnumber", referenced from:
                mlua::util::to_string::h2462b2c57fb0e3dd in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
            "_lua_isstring", referenced from:
                mlua::ffi::lua51::compat::luaL_tolstring::h2b713157f0508532 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.5.rcgu.o)
            "_lua_newthread", referenced from:
                mlua::util::protect_lua_closure::do_call::hf84c9facd2040c28 (.llvm.3734580924993591236) in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
            "_lua_newuserdata", referenced from:
                mlua::lua::Lua::create_callback::call_callback::hf62fd768a32fb2f3 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                mlua::util::protect_lua_closure::do_call::h254cc51f8553037a in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                mlua::util::protect_lua_closure::do_call::h38cf595b0c438db1 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                mlua::util::protect_lua_closure::do_call::hffb23636d71bb0ee in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                mlua::util::push_gc_userdata::h921aeedea03692a7 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                mlua::util::push_gc_userdata::h9722ae516d88946a in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                mlua::util::push_gc_userdata::he75a140f4cb19772 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                ...
            "_lua_next", referenced from:
                mlua::ffi::lua51::compat::compat53_findfield::h548718bee81eb3ed in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.5.rcgu.o)
            "_lua_objlen", referenced from:
                serde::ser::Serializer::collect_seq::h3c538280446d4782 in rsync_nvim.rsync_nvim.b3f25e6e-cgu.5.rcgu.o
                _$LT$mlua..serde..ser..SerializeVec$u20$as$u20$serde..ser..SerializeSeq$GT$::serialize_element::do_call::h5382e08d484b45ee in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.11.rcgu.o)
            "_lua_pcall", referenced from:
                mlua::function::Function::call::h20041b423bb18ae9 in rsync_nvim.rsync_nvim.b3f25e6e-cgu.11.rcgu.o
                mlua::util::protect_lua_closure::h4ebbacc47b197a8d in rsync_nvim.rsync_nvim.b3f25e6e-cgu.13.rcgu.o
                mlua::util::protect_lua_call::h5f86f7c4ef732698 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                mlua::util::protect_lua_closure::h67599b8c8d98bd27 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                mlua::util::protect_lua_closure::had1f2c3e9ee6b79d in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                mlua::util::protect_lua_closure::hb1b89abf947df66d in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                mlua::util::protect_lua_closure::hd8c3ae2370a48a20 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                ...
            "_lua_pushboolean", referenced from:
                mlua::lua::Lua::push_value::h56d0adaa5d20100c in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                mlua::util::init_gc_metatable::h4e4cc9729b215cc7 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                mlua::util::init_gc_metatable::h687303b3c509bf82 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                mlua::util::init_gc_metatable::hc3bff998e06da27a in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                mlua::util::init_error_registry::h94d046ab3d139b14 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                mlua::serde::init_metatables::do_call::hdd942496a810825a (.llvm.11307603184714879020) in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.13.rcgu.o)
            "_lua_pushcclosure", referenced from:
                mlua::function::Function::call::h20041b423bb18ae9 in rsync_nvim.rsync_nvim.b3f25e6e-cgu.11.rcgu.o
                mlua::util::protect_lua_closure::h4ebbacc47b197a8d in rsync_nvim.rsync_nvim.b3f25e6e-cgu.13.rcgu.o
                mlua::lua::Lua::create_callback::h3c79b365e6486ac4 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                mlua::lua::Lua::create_callback::do_call::ha9b98a2d49d3e02f in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                mlua::util::protect_lua_call::h5f86f7c4ef732698 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                mlua::util::protect_lua_closure::h67599b8c8d98bd27 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                mlua::util::protect_lua_closure::had1f2c3e9ee6b79d in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                ...
            "_lua_pushfstring", referenced from:
                mlua::ffi::lua51::compat::luaL_traceback::h8bc46a5d3f85f45a in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.5.rcgu.o)
                mlua::ffi::lua51::compat::luaL_tolstring::h2b713157f0508532 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.5.rcgu.o)
            "_lua_pushinteger", referenced from:
                mlua::lua::Lua::push_value::h56d0adaa5d20100c in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
            "_lua_pushlightuserdata", referenced from:
                mlua::util::protect_lua_closure::h4ebbacc47b197a8d in rsync_nvim.rsync_nvim.b3f25e6e-cgu.13.rcgu.o
                mlua::lua::Lua::init_from_ptr::h1eded340f8f0ebcb in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                mlua::lua::Lua::push_value::h56d0adaa5d20100c in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                mlua::lua::Lua::make_from_ptr::h552c3be22e898f9f in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                _$LT$mlua..lua..Lua$u20$as$u20$mlua..serde..LuaSerdeExt$GT$::array_metatable::hefd3b05a646b88f6 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                mlua::lua::Lua::init_from_ptr::_$u7b$$u7b$closure$u7d$$u7d$::do_call::h9b88f980dcf859c5 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                mlua::util::protect_lua_closure::h67599b8c8d98bd27 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                ...
            "_lua_pushlstring", referenced from:
                mlua::util::protect_lua_closure::do_call::h734df75e92650b3f (.llvm.453382158249156745) in rsync_nvim.rsync_nvim.b3f25e6e-cgu.13.rcgu.o
                mlua::lua::Lua::create_string::ha18e72160fc63138 in rsync_nvim.rsync_nvim.b3f25e6e-cgu.4.rcgu.o
                mlua::lua::Lua::create_string::hf30d545905e2b48b in rsync_nvim.rsync_nvim.b3f25e6e-cgu.4.rcgu.o
                mlua::ffi::lua51::compat::compat53_findfield::h548718bee81eb3ed in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.5.rcgu.o)
                mlua::ffi::lua51::compat::luaL_traceback::h8bc46a5d3f85f45a in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.5.rcgu.o)
                mlua::ffi::lua51::compat::luaL_tolstring::h2b713157f0508532 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.5.rcgu.o)
                mlua::util::protect_lua_closure::do_call::h2bca6393cdf8230d (.llvm.3734580924993591236) in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                ...
            "_lua_pushnil", referenced from:
                mlua::ffi::lua51::compat::compat53_findfield::h548718bee81eb3ed in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.5.rcgu.o)
                core::ptr::drop_in_place$LT$mlua..table..Table$GT$::h97f0f0ea0d3106ef in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                core::ptr::drop_in_place$LT$mlua..value..Value$GT$::h24a4d52c71741789 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                mlua::lua::Lua::push_value::h56d0adaa5d20100c in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                mlua::lua::Lua::create_callback::call_callback::hf62fd768a32fb2f3 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                mlua::lua::callback_error_ext::_$u7b$$u7b$closure$u7d$$u7d$::h4bdefdd07b1d6cfe in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                _$LT$mlua..serde..ser..SerializeMap$u20$as$u20$serde..ser..SerializeMap$GT$::end::h11e7620e646ab808 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.11.rcgu.o)
                ...
            "_lua_pushnumber", referenced from:
                mlua::lua::Lua::push_value::h56d0adaa5d20100c in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
            "_lua_pushstring", referenced from:
                mlua::serde::init_metatables::do_call::hdd942496a810825a (.llvm.11307603184714879020) in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.13.rcgu.o)
            "_lua_pushthread", referenced from:
                mlua::lua::Lua::init_from_ptr::h1eded340f8f0ebcb in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
            "_lua_pushvalue", referenced from:
                mlua::ffi::lua51::compat::lua_rotate::hdf6ac2ba506c53a9 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.5.rcgu.o)
                mlua::ffi::lua51::compat::luaL_traceback::h8bc46a5d3f85f45a in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.5.rcgu.o)
                mlua::ffi::lua51::compat::luaL_tolstring::h2b713157f0508532 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.5.rcgu.o)
                mlua::lua::Lua::push_value::h56d0adaa5d20100c in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                mlua::lua::Lua::push_ref::hc5aecdaaf7853de4 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                mlua::lua::callback_error_ext::_$u7b$$u7b$closure$u7d$$u7d$::h4bdefdd07b1d6cfe in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                mlua::util::rawset_field::h1a4892e2eba80cce in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                ...
            "_lua_rawequal", referenced from:
                mlua::ffi::lua51::compat::compat53_findfield::h548718bee81eb3ed in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.5.rcgu.o)
                mlua::util::pop_error::h04ae61cb9f92798f in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                mlua::util::get_gc_userdata::hb89e34bf79757476 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                mlua::util::error_traceback::he68037b9b5234d0f in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                std::panicking::try::hc905dbe356661666 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.12.rcgu.o)
            "_lua_rawget", referenced from:
                mlua::lua::Lua::init_from_ptr::h1eded340f8f0ebcb in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                mlua::lua::Lua::make_from_ptr::h552c3be22e898f9f in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                _$LT$mlua..lua..Lua$u20$as$u20$mlua..serde..LuaSerdeExt$GT$::array_metatable::hefd3b05a646b88f6 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                mlua::util::push_gc_userdata::h9722ae516d88946a in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                mlua::util::push_gc_userdata::hcecbf9ac4cbf0473 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                mlua::util::push_gc_userdata::he75a140f4cb19772 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                mlua::util::userdata_destructor::h04ee7e5326c9d154 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                ...
            "_lua_rawset", referenced from:
                mlua::table::Table::set::hff356e86ce69b428 in rsync_nvim.rsync_nvim.b3f25e6e-cgu.3.rcgu.o
                mlua::table::Table::raw_set::hd238aea89fcbe55e in rsync_nvim.rsync_nvim.b3f25e6e-cgu.3.rcgu.o
                mlua::table::Table::raw_set::hd3243c56630bce63 in rsync_nvim.rsync_nvim.b3f25e6e-cgu.3.rcgu.o
                mlua::lua::Lua::init_from_ptr::_$u7b$$u7b$closure$u7d$$u7d$::do_call::h9b88f980dcf859c5 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                mlua::util::protect_lua_closure::do_call::h2bca6393cdf8230d (.llvm.3734580924993591236) in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                mlua::util::protect_lua_closure::do_call::h418a40b0275da235 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                mlua::util::init_error_registry::do_call::hfc47759815fbee4e in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                ...
            "_lua_rawseti", referenced from:
                serde::ser::Serializer::collect_seq::h3c538280446d4782 in rsync_nvim.rsync_nvim.b3f25e6e-cgu.5.rcgu.o
                _$LT$mlua..serde..ser..SerializeVec$u20$as$u20$serde..ser..SerializeSeq$GT$::serialize_element::do_call::h5382e08d484b45ee in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.11.rcgu.o)
            "_lua_remove", referenced from:
                mlua::util::protect_lua_closure::h4ebbacc47b197a8d in rsync_nvim.rsync_nvim.b3f25e6e-cgu.13.rcgu.o
                mlua::ffi::lua51::compat::compat53_findfield::h548718bee81eb3ed in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.5.rcgu.o)
                mlua::ffi::lua51::compat::luaL_traceback::h8bc46a5d3f85f45a in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.5.rcgu.o)
                mlua::lua::Lua::create_callback::call_callback::hf62fd768a32fb2f3 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                mlua::util::protect_lua_call::h5f86f7c4ef732698 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                mlua::util::protect_lua_closure::h67599b8c8d98bd27 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                mlua::util::protect_lua_closure::had1f2c3e9ee6b79d in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                ...
            "_lua_replace", referenced from:
                mlua::ffi::lua51::compat::lua_rotate::hdf6ac2ba506c53a9 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.5.rcgu.o)
                mlua::ffi::lua51::compat::luaL_traceback::h8bc46a5d3f85f45a in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.5.rcgu.o)
                mlua::ffi::lua51::compat::luaL_tolstring::h2b713157f0508532 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.5.rcgu.o)
                core::ptr::drop_in_place$LT$mlua..table..Table$GT$::h97f0f0ea0d3106ef in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                core::ptr::drop_in_place$LT$mlua..value..Value$GT$::h24a4d52c71741789 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                mlua::lua::Lua::push_value::h56d0adaa5d20100c in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                mlua::lua::Lua::create_callback::call_callback::hf62fd768a32fb2f3 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                ...
            "_lua_setmetatable", referenced from:
                mlua::lua::Lua::create_callback::call_callback::hf62fd768a32fb2f3 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                mlua::util::push_gc_userdata::h921aeedea03692a7 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                mlua::util::push_gc_userdata::h9722ae516d88946a in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                mlua::util::push_gc_userdata::hcecbf9ac4cbf0473 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                mlua::util::push_gc_userdata::he75a140f4cb19772 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                mlua::util::userdata_destructor::h04ee7e5326c9d154 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                mlua::util::userdata_destructor::h13b3004672793de7 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                ...
            "_lua_settable", referenced from:
                mlua::table::Table::set::do_call::he6a5adc4e6e75ca4 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.3.rcgu.o)
            "_lua_settop", referenced from:
                mlua::function::Function::call::h20041b423bb18ae9 in rsync_nvim.rsync_nvim.b3f25e6e-cgu.11.rcgu.o
                mlua::util::protect_lua_closure::do_call::h734df75e92650b3f (.llvm.453382158249156745) in rsync_nvim.rsync_nvim.b3f25e6e-cgu.13.rcgu.o
                mlua::table::Table::set::hff356e86ce69b428 in rsync_nvim.rsync_nvim.b3f25e6e-cgu.3.rcgu.o
                mlua::table::Table::raw_set::hd238aea89fcbe55e in rsync_nvim.rsync_nvim.b3f25e6e-cgu.3.rcgu.o
                mlua::table::Table::raw_set::hd3243c56630bce63 in rsync_nvim.rsync_nvim.b3f25e6e-cgu.3.rcgu.o
                serde::ser::Serializer::collect_seq::h3c538280446d4782 in rsync_nvim.rsync_nvim.b3f25e6e-cgu.5.rcgu.o
                mlua::ffi::lua51::compat::compat53_findfield::h548718bee81eb3ed in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.5.rcgu.o)
                ...
            "_lua_toboolean", referenced from:
                mlua::ffi::lua51::compat::luaL_tolstring::h2b713157f0508532 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.5.rcgu.o)
                mlua::lua::Lua::pop_value::h6ff2b899fb65a0e7 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                mlua::util::to_string::h2462b2c57fb0e3dd in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
            "_lua_tolstring", referenced from:
                mlua::util::protect_lua_closure::do_call::h734df75e92650b3f (.llvm.453382158249156745) in rsync_nvim.rsync_nvim.b3f25e6e-cgu.13.rcgu.o
                mlua::lua::Lua::create_string::ha18e72160fc63138 in rsync_nvim.rsync_nvim.b3f25e6e-cgu.4.rcgu.o
                mlua::lua::Lua::create_string::hf30d545905e2b48b in rsync_nvim.rsync_nvim.b3f25e6e-cgu.4.rcgu.o
                mlua::ffi::lua51::compat::luaL_traceback::h8bc46a5d3f85f45a in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.5.rcgu.o)
                mlua::ffi::lua51::compat::luaL_tolstring::h2b713157f0508532 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.5.rcgu.o)
                mlua::lua::Lua::coerce_string::h1d3c85814117ca21 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                mlua::util::protect_lua_closure::do_call::h2bca6393cdf8230d (.llvm.3734580924993591236) in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                ...
            "_lua_tonumber", referenced from:
                mlua::lua::Lua::pop_value::h6ff2b899fb65a0e7 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                mlua::util::to_string::h2462b2c57fb0e3dd in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
            "_lua_topointer", referenced from:
                mlua::ffi::lua51::compat::luaL_tolstring::h2b713157f0508532 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.5.rcgu.o)
                mlua::lua::Lua::init_from_ptr::h1eded340f8f0ebcb in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                mlua::util::get_gc_userdata::hb89e34bf79757476 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                mlua::util::to_string::h2462b2c57fb0e3dd in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
            "_lua_touserdata", referenced from:
                mlua::util::protect_lua_closure::do_call::h734df75e92650b3f (.llvm.453382158249156745) in rsync_nvim.rsync_nvim.b3f25e6e-cgu.13.rcgu.o
                mlua::lua::Lua::pop_value::h6ff2b899fb65a0e7 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                mlua::lua::Lua::create_callback::call_callback::hf62fd768a32fb2f3 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                mlua::lua::Lua::make_from_ptr::h552c3be22e898f9f in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                mlua::lua::callback_error_ext::_$u7b$$u7b$closure$u7d$$u7d$::h4bdefdd07b1d6cfe in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                mlua::util::protect_lua_closure::do_call::h254cc51f8553037a in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                mlua::util::protect_lua_closure::do_call::h2bca6393cdf8230d (.llvm.3734580924993591236) in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.0.rcgu.o)
                ...
            "_lua_type", referenced from:
                mlua::ffi::lua51::compat::compat53_findfield::h548718bee81eb3ed in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.5.rcgu.o)
                mlua::ffi::lua51::compat::luaL_tolstring::h2b713157f0508532 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.5.rcgu.o)
                mlua::lua::Lua::init_from_ptr::h1eded340f8f0ebcb in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                mlua::lua::Lua::pop_value::h6ff2b899fb65a0e7 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                mlua::lua::Lua::create_callback::call_callback::hf62fd768a32fb2f3 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                mlua::lua::Lua::make_from_ptr::h552c3be22e898f9f in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                _$LT$mlua..lua..Lua$u20$as$u20$mlua..serde..LuaSerdeExt$GT$::array_metatable::hefd3b05a646b88f6 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                ...
            "_lua_typename", referenced from:
                mlua::ffi::lua51::compat::luaL_tolstring::h2b713157f0508532 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.5.rcgu.o)
            "_lua_xmove", referenced from:
                mlua::lua::Lua::create_table_with_capacity::h5894c7c379a161e3 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                mlua::lua::Lua::coerce_string::h1d3c85814117ca21 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                mlua::lua::Lua::push_value::h56d0adaa5d20100c in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                mlua::lua::Lua::pop_value::h6ff2b899fb65a0e7 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                mlua::lua::Lua::push_ref::hc5aecdaaf7853de4 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                mlua::lua::Lua::pop_ref::hf516c9438cb45e67 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                mlua::lua::Lua::create_callback::h3c79b365e6486ac4 in libmlua-76ae36068b5adf9d.rlib(mlua-76ae36068b5adf9d.mlua.02e97f32-cgu.1.rcgu.o)
                ...
          ld: symbol(s) not found for architecture arm64
          clang: error: linker command failed with exit code 1 (use -v to see invocation)


error: could not compile `rsync-nvim` (lib) due to previous error
make: *** [build] Error 101
```

--- a/.github/workflows/busted.yml
+++ b/.github/workflows/busted.yml
@@ -33,7 +33,7 @@ jobs:
           repository: nvim-lua/plenary.nvim
           path: plenary.nvim
       - name: install luacov
-        if: ${{ matix.is_unix }}
+        if: ${{ matrix.is_unix }}
         run: |
           sudo apt install luarocks
           sudo luarocks install luacov

--- a/.github/workflows/busted.yml
+++ b/.github/workflows/busted.yml
@@ -49,7 +49,10 @@ jobs:
         run: NOCLEAN=1 make testcov
       - name: run on macos
         if: ${{ matrix.is_mac }}
-        run: make test
+        run: |
+          # to get latest version of rsync
+          brew install rsync
+          make test
       - name: upload coverage report
         if: ${{ matrix.is_unix }}
         uses: codecov/codecov-action@v3

--- a/.github/workflows/busted.yml
+++ b/.github/workflows/busted.yml
@@ -7,8 +7,16 @@ on:
 
 jobs:
   busted:
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+        include:
+          - os: macos-latest
+            is_mac: 1
+          - os: ubuntu-latest
+            is_unix: 1
     name: busted
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
     steps: 
       - name: checkout
         uses: actions/checkout@v3
@@ -25,6 +33,7 @@ jobs:
           repository: nvim-lua/plenary.nvim
           path: plenary.nvim
       - name: install luacov
+        if: ${{ matix.is_unix }}
         run: |
           sudo apt install luarocks
           sudo luarocks install luacov
@@ -36,8 +45,13 @@ jobs:
           neovim: true
           version: nightly
       - name: run test
+        if: ${{ matrix.is_unix }}
         run: NOCLEAN=1 make testcov
+      - name: run on macos
+        if: ${{ matrix.is_mac }}
+        run: make test
       - name: upload coverage report
+        if: ${{ matrix.is_unix }}
         uses: codecov/codecov-action@v3
         with:
           fail_ci_if_error: true

--- a/Makefile
+++ b/Makefile
@@ -59,6 +59,8 @@ testcov-html:
 .PHONY: test-clean
 test-clean:
 	@rm -rf luacov*
+	@rm -rf lua/deps
+	@rm lua/rsync_nvim.so
 
 .PHONY: clean
 clean: test-clean

